### PR TITLE
FIX commonJS browserify support

### PR DESCRIPTION
**Pull request by serapath**
my editor also fixed some trailing white space issues - hope thats ok.

### CODE REVIEW

**Comment by marksyzm** [comment]
Ahhh I really hope this goes in soon! Just working on a large commercial project where we're about to pay for support and I'm switching over to use browserify. I'll use your repo for now until this gets merged as I'm struggling to get this all connected on the 8.0 version with browserify shims.

**Comment by marksyzm** [comment]
@serapath So close! Manhattan joints not working: 
```
TypeError: Cannot read property 'manhattan' of undefined
    at joint.dia.LinkView.joint.dia.CellView.extend.findRoute (joint.dia.link.js:870)
    at joint.dia.LinkView.joint.dia.CellView.extend.update (joint.dia.link.js:463)
    at bound [as update] (lodash.js:1318)
    at joint.dia.LinkView.joint.dia.CellView.extend.render (joint.dia.link.js:277)
    at joint.dia.Paper.Backbone.View.extend.addCell (joint.dia.paper.js:372)
    at bound [as addCell] (lodash.js:1318)
    at joint.dia.Paper.Backbone.View.extend.onAddCell (joint.dia.paper.js:362)
    at triggerEvents (backbone.js:209)
    at Backbone.Events.trigger (backbone.js:148)
    at triggerEvents (backbone.js:210)
```
I'll see if I can fork your branch and work out the issue, unless you get there first...

**Comment by marksyzm** [comment]
Fixed: https://github.com/serapath/joint/pull/1

**Comment by serapath** [comment]
thx @marksyzm :-) i hope this gets merged back into the original project. For now i'm using my fork.
You can install it using `npm install serapath/joint --save` instead of `npm install daviddurman/joint`. The `npm install jointjs` is outdated anyway and still uses version 0.8.0

**Comment by marksyzm** [comment]
Yeah, I was trying it out and thought "okay, I'll just fork it and... oh, there's all this stuff in here for the 9.3.0 version..." and was stumped for a bit because of all the algorithm stuff, then found this pr! Brilliant. Yeah this looks okay to me for a release to 9.3.1 but what do I know :)

**Comment by dlcwalkoff** [comment]
And need add support all plugins. For example DirectedGraph.

**Comment by marksyzm** [comment]
​I wonder if that should/could be a separate `require`

**Comment by dylanvorster** [comment]
+1 For it all working first time. Tried a number of other forked repos that tried to do a similar thing, and this really did the trick. That said, whats still preventing this from being merged in? The plugins? @dlcwalkoff has also done some nice work trying to clean it up. His remaster branch keeps things simple and leaves the packaging up to the end user.

**Comment by serapath** [comment]
i dont know what keeps it from beeing merged.
the repo owner said he would take a look. i guess he is busy.

**Comment by kumilingus** [comment]
The solution introduced by this PR is breaking the JointJS to run in NodeJS environment. (try `make stest`).

PS: we are working on this issue and should be able to release it in a week or two.

File: index.js [hunk 1]
```diff
@@ -1,14 +1,42 @@
...
+  dia:    {
+    Cell:     {},
+    CellView: {},
+    Graph:    {},
+    Link:     {},
+    Element:  {},
+    Paper:    {}
+  }
 };
+var jointjs = require('jointjs');
```
Comment (kumilingus) [inline index.js]: Is this actually working for you? Doesn't seem to me that you are exporting the `jointjs` library here (but the namespaces only).

**Comment by serapath** [comment]
puh, @marksyzm what exactly did you change? You removed a lot of code that was added to make the browserify thing work. Did you do the changes ON TOP of my fork or the original fork and then just created a PR to my fork?

@kumilingus maybe you can check out HEAD~2, the commit is called "FIX commonJS browserify support" and remove everything that comes after. Does it then work for you?

**Comment by marksyzm** [comment]
I didn't remove anything? Do you see any red lines here? https://github.com/serapath/joint/commit/f3f6ed57a2019d1a69d5f8ecd90d3f58d04ff104

**Comment by marksyzm** [comment]
I would suggest we wait for them to fix the issue themselves, anyway. They've got a lot of stuff going on. only changed this to fix my needs but I was also using rappid and that has so many issues with browserify that I just ended up using jointjs and rappid as a global.

**Comment by kumilingus** [comment]
Duplicate of #135.

**Pull request closed without merging**